test/e2e/apps/disruption.go

```diff
@@ -372,9 +373,9 @@ func locateRunningPod(cs kubernetes.Interface, ns string) (pod *v1.Pod, err erro
 			return false, err
 		}
 
-		for i := range podList.Items {
-			if podList.Items[i].Status.Phase == v1.PodRunning {
-				pod = &podList.Items[i]
+		for _, p := range podList.Items {
+			if podutil.IsPodReady(&p) {
+				pod = &p
```

Comment:
is this returning address of a temporary variable?

Reply:
Yeah, it is taking the address of the iteration variable. It works in this situation due to the early return after the assignment, but we should avoid this. I have updated the PR.